File: build_lib/ipynb.mjs
const fs = require('fs')

function strip_ipynb(file) {
  const nb = JSON.parse(fs.readFileSync(file, 'utf8'))
  for (const cell of nb.cells) {
    if (cell.cell_type === 'code') {
      cell.outputs = []
      cell.execution_count = null
    }
  }

  fs.writeFileSync(file, JSON.stringify(nb, null, 2))
}

export default strip_ipynb

File: build_py/src/build_py/ipynb.py
import json
from pathlib import Path


def ipynb(path: str | Path):
    path = Path(path)
    print(f"IPYNB {path.resolve()}")
    with open(path, encoding="utf8") as file:
        nb = json.load(file)
    for cell in nb["cells"]:
        if cell["cell_type"] == "code":
            cell["outputs"] = []
            cell["execution_count"] = None
    with open(path, "w", encoding="utf8") as file:
        json.dump(nb, file, ensure_ascii=False, indent=2)
